server/src/main/java/org/opensearch/action/admin/cluster/remotestore/stats/RemoteStoreStatsResponse.java

```diff
@@ -41,30 +50,85 @@ public RemoteStoreStatsResponse(
         List<DefaultShardOperationFailedException> shardFailures
     ) {
         super(totalShards, successfulShards, failedShards, shardFailures);
-        this.shards = shards;
+        this.remoteStoreStats = shards;
     }
 
-    public RemoteStoreStats[] getShards() {
-        return this.shards;
+    public RemoteStoreStats[] getRemoteStoreStats() {
+        return this.remoteStoreStats;
     }
 
     public RemoteStoreStats getAt(int position) {
-        return shards[position];
+        return remoteStoreStats[position];
+    }
+
+    public Map<String, Map<Integer, List<RemoteStoreStats>>> groupByIndexAndShards() {
```

Comment:
Why can't this method return the map of map directly? Will the `this.indexWiseStats` be used even after the method `protected void addCustomXContentFields(XContentBuilder builder, Params params) throws IOException {`?
If not, we can directly return the map of map, and use it.

Reply:
Will address this